lib/router/with-router.js

```diff
@@ -23,5 +19,10 @@ export default function withRouter (ComposedComponent) {
     }
   }
 
+  if (process.env.NODE_ENV === 'development') {
```

Comment:
I would make this `process.env.NODE_ENV !== 'production'` instead. Jest uses the "test" environment for example and this info might be useful for React warnings and error stacks there.